src/util/invariant.h

```diff
@@ -160,10 +163,17 @@ invariant_violated_structured(
   const std::string &file,
   const std::string &function,
   const int line,
+  const std::string &condition,
   Params &&... params)
 {
   std::string backtrace=get_backtrace();
-  ET to_throw(file, function, line, backtrace, std::forward<Params>(params)...);
+  ET to_throw(
+    file,
+    function,
+    line,
+    backtrace,
+    std::forward<Params>(params)...,
+    condition);
```

Comment:
IMHO patches are easier to read if whitespace changes to existing code are in a different commit.